src/services/productService.ts

```diff
@@ -70,8 +83,14 @@ export class ProductService {
                 results.push(newProduct);
             })
             .on('end', () => {
-                const combinedData = [...existingData, ...results];
-                fs.writeFileSync(PRODUCTS_FILE_PATH, JSON.stringify(combinedData, null, 2)); // Перезаписуємо весь масив
+                results.forEach((product, index) => {
```

Comment:
Since you write to the file once, only at the end of the `csv()` stream, the memory consumption problem still exists